### ISSUE TITLE
Difference loss by model.evaluate and model.predict with kernel_regularizer option

### ISSUE THREAD

**Issue by ht93** [keras-team/keras#8242]
### Check box
- [x] Check that you are up-to-date with the master branch of Keras. 
    Keras version: 2.0.8

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. 
     tensorflow version: 1.3.0 

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

### Issue
* Code: [gist1](https://gist.github.com/ht93/05f07c4610df21964324e92555e00223#file-gist1-py) [gist2](https://gist.github.com/ht93/05f07c4610df21964324e92555e00223#file-gist2-py)
* Issue: Both `gist1` and `gist2` are two `Dense` layer model without any BatchNorm and Dropout. The only difference between two model is that `gist2` has `kernel_regularizer=regularizers.l2(0.02)` in first `Dense` layer. I calculated the MSE loss after each epoch with `model.evaluate` and `model.predict`. The MSE loss for `model.predict` is calculated by numpy `np.mean(np.square(train_y - y_predict), axis=-1)`. For `gist1` model, two MSE losses are same.  For `gist2` model, two MSE losses are different.
* Output: 
```
# gist1
Epoch: 0001     train_loss= 0.5504                          valid_loss= 0.2229
After training: train_loss= 0.2316 train_loss_numpy= 0.2316 valid_loss= 0.2229 valid_loss_numpy= 0.2229
Epoch: 0002     train_loss= 0.1728                          valid_loss= 0.1623
After training: train_loss= 0.1607 train_loss_numpy= 0.1607 valid_loss= 0.1623 valid_loss_numpy= 0.1623
Epoch: 0003     train_loss= 0.1588                          valid_loss= 0.1447
After training: train_loss= 0.1427 train_loss_numpy= 0.1427 valid_loss= 0.1447 valid_loss_numpy= 0.1447
```
```
# gist2
Epoch: 0001     train_loss= 2.2506                          valid_loss= 2.0081
After training: train_loss= 2.0028 train_loss_numpy= 0.1761 valid_loss= 2.0081 valid_loss_numpy= 0.1814
Epoch: 0002     train_loss= 1.8969                          valid_loss= 1.7843
After training: train_loss= 1.7805 train_loss_numpy= 0.1294 valid_loss= 1.7843 valid_loss_numpy= 0.1333
Epoch: 0003     train_loss= 1.6961                          valid_loss= 1.5916
After training: train_loss= 1.5888 train_loss_numpy= 0.1191 valid_loss= 1.5916 valid_loss_numpy= 0.1219
```
